trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmCrossImporter.php

```diff
@@ -0,0 +1,972 @@
+   description: new TranslatableMarkup('Loads germplasm crosses into the system. This is useful for large datasets to ease the upload process.'),
+   file_types: ['tsv'],
+   upload_description: new TranslatableMarkup('Please provide a data file.'),
+   upload_title: new TranslatableMarkup('Germplasm Cross Data File*'),
+   use_analysis: FALSE,
+   require_analysis: FALSE,
+   use_button: TRUE,
+   submit_disabled: FALSE,
+   button_text: new TranslatableMarkup('Import'),
+   file_upload: TRUE,
+   file_local: FALSE,
+   file_remote: FALSE,
+   file_required: TRUE,
+   cardinality: 1,
+   menu_path: '',
+   callback: '',
+   callback_path: '',
+  )]
+class GermplasmCrossImporter extends ChadoImporterBase implements ContainerFactoryPluginInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * Headers required by this importer.
+   *
+   * @var array
+   *
+   * The following keys are required:
+   * - 'name': The column header name as it should appear in the input file.
+   * - 'description': A user-friendly description of the header that will be
+   *   displayed to the user through the form.
+   * - 'type': one of "required" or "optional" to indicate whether the column
+   *   needs to have values present or not.
+   *
+   * NOTE: Order MUST reflect the desired order of headers in the input file.
+   */
+  private array $headers = [
+    [
+      'name' => 'Year',
+      'description' => 'The year this cross was made in (e.g. 2020).',
+      'type' => 'required',
+    ],
+    [
+      'name' => 'Season',
+      'description' => 'The season this cross was made in (one of: Spring, Fall, Winter, Summer).',
+      'type' => 'required',
+    ],
+    [
+      'name' => 'Cross Number',
+      'description' => 'The generic name for this cross (e.g. 1234S).',
+      'type' => 'required',
+    ],
+    [
+      'name' => 'Uniquename',
+      'description' => 'A unique identifier for this cross. This can be the same as Cross Number, if desired.',
+      'type' => 'required',
+    ],
+    [
+      'name' => 'Maternal Parent',
+      'description' => 'The name of the maternal parent of this cross.',
+      'type' => 'required',
+    ],
+    [
+      'name' => 'Paternal Parent',
+      'description' => 'The name of the paternal parent of this cross.',
+      'type' => 'required',
+    ],
+    [
+      'name' => 'Cross Type',
+      'description' => 'The type of cross (e.g. single, double, triple).',
+      'type' => 'required',
+    ],
+    [
+      'name' => 'Seed Type',
+      'description' => 'Either the market class or the seed coat colour of the seed resulting from this cross.',
+      'type' => 'optional',
+    ],
+    [
+      'name' => 'Cotyledon Colour',
+      'description' => 'The cotyledon colour of the seed resulting from this cross.',
+      'type' => 'optional',
+    ],
+    [
+      'name' => 'Comment',
+      'description' => 'A free-text comment about this cross.',
+      'type' => 'optional',
+    ],
```

Comment:
Eventually we are going to need a lot more flexibility in the metadata columns in these importers. However, that is beyond the scope of this PR so I created issue https://github.com/TripalCultivate/TripalCultivate-Germplasm/issues/43 to capture this future need.